### ISSUE TITLE
`Struct.new do` and `Metrics/BlockLength` vs `Metrics/ClassLength`

### ISSUE THREAD

**Issue by marcandre** [rubocop-hq/rubocop#8102]
```ruby
# This style is used by many people
MyClass < Struct.new(:foo, :bar)
  # ...
end

# This style is mostly equivalent and is the "official" style:
MyClass = Struct.new(:foo, :bar) do
  # ...
end
```

Both styles should give the same offenses.

Currently `Metrics/BlockLength` will complain (when this is really not a standard block at all) and `Metrics/ClassLength` will not take it into account (when it should).

~~If we don't have a cop for to favor one style over the other, maybe we should~~ (we do:  Style/StructInheritance)
